src/nodes/dir.rs

```diff
@@ -16,14 +16,34 @@ extern crate fuse;
 extern crate time;
 
 use {Cache, IdGenerator};
+use failure::{Error, ResultExt};
 use nix::{errno, unistd};
 use nodes::{KernelError, Node, NodeResult, conv};
 use std::collections::HashMap;
 use std::ffi::{OsStr, OsString};
 use std::fs;
-use std::path::{Path, PathBuf};
+use std::io;
+use std::path::{Component, Path, PathBuf};
 use std::sync::{Arc, Mutex};
-use time::Timespec;
+
+/// Takes the components of a path and returns the first normal component and the rest.
+///
+/// This assumes that the input path is normalized and that the very first component is a normal
+/// component as defined by `Component::Normal`.
+fn split_components<'a>(components: &'a [Component]) -> (&'a OsStr, &'a [Component<'a>]) {
```

Comment:
One is `Component`, the other `Component<'a>`. Can you make them match? Can you remove all the `'a` specifiers, since they're all the same?

Reply:
Made them match. I had to add these as otherwise the compiler complains for both return values with errors like:

```
error[E0106]: missing lifetime specifier
  --> src/nodes/dir.rs:33:51
   |
33 | fn split_components(components: &[Component]) -> (&OsStr, &[Component]) {
   |                                                   ^ expected lifetime parameter
   |
   = help: this function's return type contains a borrowed value, but the signature does not say which one of `components`'s 2 lifetimes it is borrowed from
```